### ISSUE TITLE
Meteobridge flips battery status with ambientweather

### ISSUE THREAD

**Issue by staze** [themoosman/weewx-ambientweatherapi-json#18]
As requested. =)

https://github.com/ambient-weather/api-docs/wiki/Device-Data-Specs

battout - OK/Low indication, Int, 1=OK, 0=Low (Meteobridge Users 1=Low, 0=OK)
batt1...batt10 - OK/Low indication, Int, 1=OK, 0=Low (Meteobridge Users 1=Low, 0=OK)
batt_25 - PM2.5 Air Quality Sensor Battery indication, OK/Low indication, Int, 1=OK, 0=Low (Meteobridge Users 1=Low, 0=OK)

From an api test call, I can confirm:

      "battin": 0,
      "battout": 0,

Adding a setting to flip this logic would be great. 

Thanks!

**Comment by themoosman** [comment]
@staze I tested as much as I could without a Meteobridge, can you check out the branch and test for me?

https://github.com/themoosman/weewx-ambientweatherapi-json/tree/issue-18

**Comment by staze** [comment]
Testing now. Interestingly, I had grabbed the last main branch last week and noticed that it was showing different results after that update. Outside showed "Okay" and Inside showed "Low". 

Waiting for a web refresh to see what this shows now... 

Okay, now, ironically, it shows the opposite. Outside shows Low, Inside shows "OK"

Maybe I misunderstood the setting in weewx.conf. I did set it to "use_meteobridge = TRUE" Maybe True needs to be in quotes?

**Comment by themoosman** [comment]
True does not need to be in quotes.  It will print the setting in the log, look for an entry like `use_meteobridge: True`

**Comment by themoosman** [comment]
You can also set `debug = 1` in `weewx.conf` to get more information.

**Comment by themoosman** [comment]
You also might have to post an API response JSON.  AW does not report an "inside" battery status, as it nests them in `bat1`, `bat2`, etc.

**Comment by staze** [comment]
Debug shows:

```
971:Mar  8 13:48:37 raspi-server-misc weewx[23838] INFO __main__: Terminating weewx version 4.4.0
973:Mar  8 13:48:38 raspi-server-misc weewx[29972] INFO __main__: Initializing weewx version 4.4.0
974:Mar  8 13:48:38 raspi-server-misc weewx[29972] INFO __main__: Using Python 3.7.3 (default, Jul 25 2020, 13:03:44) #012[GCC 8.3.0]
975:Mar  8 13:48:38 raspi-server-misc weewx[29972] INFO __main__: Platform Linux-5.10.11-v7l+-armv7l-with-debian-10.8
976:Mar  8 13:48:38 raspi-server-misc weewx[29972] INFO __main__: Locale is 'en_US.UTF-8'
977:Mar  8 13:48:38 raspi-server-misc weewx[29972] INFO __main__: PID file is /var/run/weewx.pid
978:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO __main__: Using configuration file /etc/weewx/weewx.conf
979:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO __main__: Debug is 1
980:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.engine: Loading station type ambientweatherapi (user.ambientweatherapi)
981:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO root: use_meteobridge: True
982:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO root: Starting: ambientweatherapi, version: 0.3
984:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.engine: StdConvert target unit is 0x1
986:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.engine: Archive will use data binding wx_binding
987:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.engine: Record generation will be attempted in 'hardware'
988:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.engine: Using archive interval of 300 seconds (specified in weewx configuration)
989:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.restx: StationRegistry: Registration not requested.
990:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.restx: Wunderground: Posting not enabled.
991:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.restx: PWSweather: Posting not enabled.
992:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.restx: CWOP: Posting not enabled.
993:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.restx: WOW: Posting not enabled.
994:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.restx: AWEKAS: Posting not enabled.
995:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO __main__: Starting up weewx version 4.4.0
996:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.engine: Using binding 'wx_binding' to database 'weewx.sdb'
997:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.manager: Starting backfill of daily summaries
998:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.manager: Daily summaries up to date
999:Mar  8 13:48:38 raspi-server-misc weewx[29976] INFO weewx.engine: Starting main packet loop.
1173:Mar  8 13:49:39 raspi-server-misc weewx[29976] INFO weewx.wxxtypes: Type beaufort has been deprecated. Use unit beaufort instead.
1174:Mar  8 13:49:39 raspi-server-misc weewx[29976] INFO root: loopPacket Accepted
1373:Mar  8 13:50:42 raspi-server-misc weewx[29976] INFO root: loopPacket Accepted
```

API data:

```
[
  {
    "macAddress": "xxxxxxxx",
    "lastData": {
      "dateutc": 1615240500000,
      "winddir": 180,
      "windspeedmph": 0,
      "windgustmph": 0.9,
      "maxdailygust": 20,
      "windgustdir": 220,
      "winddir_avg2m": 180,
      "windspdmph_avg2m": 0,
      "winddir_avg10m": 65,
      "windspdmph_avg10m": 0,
      "tempf": 49.3,
      "humidity": 61,
      "baromrelin": 29.93,
      "baromabsin": 29.42,
      "tempinf": 73.6,
      "humidityin": 34,
      "hourlyrainin": 0,
      "dailyrainin": 0,
      "monthlyrainin": 0.7,
      "yearlyrainin": 12.04,
      "battin": 0,
      "battout": 1,
      "feelsLike": 49.3,
      "dewPoint": 36.43,
      "feelsLikein": 72.3,
      "dewPointin": 43.5,
      "lastRain": "2021-03-07T12:55:00.000Z",
      "tz": "America/Los_Angeles",
      "date": "2021-03-08T21:55:00.000Z"
    },
    "info": {
      "name": "Home",
      "location": "Home",
      "coords": {
        "coords": {
          "lat": 44.05120759999999,
          "lon": -123.0365696
        },
        "address": "xxxxx",
        "location": "Springfield",
        "elevation": 136.5034027099609,
        "geo": {
          "type": "Point",
          "coordinates": [
            -123.xxxx,
            44.xxxxx
          ]
        }
      }
    }
  }
]
```

**Comment by themoosman** [comment]
```
 "battin": 0,
 "battout": 1,
 ```
 I wonder if `battin` follows the default or meteobridge method for reporting?  AW does not document that setting.

**Comment by staze** [comment]
not sure. as mentioned earlier, prior to this branch, battin was showing as "okay" I believe. 

I imagine since they mention meteobridge flips the bit on all the others, it would on this one as well.

**Comment by themoosman** [comment]
I opened an issue https://github.com/ambient-weather/api-docs/issues/29

**Comment by themoosman** [comment]
> not sure. as mentioned earlier, prior to this branch, battin was showing as "okay" I believe.
> 
> I imagine since they mention meteobridge flips the bit on all the others, it would on this one as well.

Yeah, so the flip logic in this code is working :)

**Comment by staze** [comment]
Though, that's odd. Cause as far as I know, the batteries on both are fine... 

Let me take a closer look at my setup and see if the battery is in fact low.

**Comment by staze** [comment]
Yup, my meteobridge is reporting a low battery. Guess I know what I'm doing this evening. lol

**Comment by staze** [comment]
ha. it just went low today. figures...

**Comment by themoosman** [comment]
Of course, it did.  :laughing:

**Comment by staze** [comment]
![Battery Status](https://user-images.githubusercontent.com/1100483/110390225-8099cc00-801a-11eb-9d6c-233c006789ce.jpg)

I'm not kidding! what are the odds. lol

**Comment by themoosman** [comment]
Let me know if that other battery reports good and I'll merge the code.

**Comment by staze** [comment]
I believe you should be good to go. The console battery should always be good since it's plugged into power. 

So yup, go ahead and merge. I'll go get the station battery replaced tonight and report back if that resolves the low one.

**Comment by themoosman** [comment]
@staze Did the new battery resolve the alert?

**Comment by staze** [comment]
Replaced battery yesterday, console is still showing low battery. Normally it should reset at midnight and show it's fine, but it didn't. So either the battery I installed was already low (unlikely) or I'm seeing the occasion where it takes a couple days. 

https://support.davisinstruments.com/article/bv6nzyga1g-i-m-getting-a-low-battery-transmitter-1-error-on-my-console

Will post back once I have an answer.
